src/Client/Client.php

```diff
@@ -79,6 +95,12 @@ public function handleEvent($event, $name = null)
             }
         }
 
+        if (null !== $this->toSendLimit && $this->getToSend()->count() >= $this->toSendLimit) {
```

Comment:
There is no reason to use both `immediate_send` and `to_send_limit` config so I think we can let this test.